### PR TITLE
Per guild stuff.

### PR DESCRIPTION
**Pull request by oliver-ni**
Moving stuff to be generalizable.

### CODE REVIEW

File: cogs/logging.py [hunk 1]
```diff
@@ -91,13 +83,14 @@ async def on_guild_updates(self, *args):
...
-        if isinstance(thing, discord.User):
-            guild = self.bot.get_guild(GUILD_ID)
-            thing = guild.get_member(thing.id)
-        await self.sync_member(thing)
+        await self.sync_member(args[-1])
+
+    @commands.Cog.listener()
+    async def on_user_update(self, *args):
+        for guild in self.bot.guilds:
+            member = guild.get_member(args[-1].id)
```
Comment (pikaninja) [inline cogs/logging.py]: what if member is None?

File: cogs/moderation.py [hunk 2]
```diff
@@ -284,7 +281,7 @@ async def convert(self, ctx, arg):
 
 
 class MemberOrIdConverter(commands.Converter):
-    async def convert(self, ctx, arg):
+    async def convert(self, ctx, arg) -> discord.Member:
```
Comment (pikaninja) [inline cogs/moderation.py]: why random typehint here?

File: cogs/moderation.py [hunk 3]
```diff
@@ -654,6 +683,7 @@ async def tradingmute(
             target=target,
             user=ctx.author,
             reason=reason,
+            guild_id=ctx.guild.id,
```
Comment (pikaninja) [inline cogs/moderation.py:682]: This *might* break on new servers? You have setup for muted but no setup or anythign for trading mute/trading

File: cogs/names.py [hunk 4]
```diff
@@ -42,14 +41,12 @@ async def on_member_update(self, before, after):
     async def on_user_update(self, before, after):
         if before.name == after.name:
             return
-        guild = self.bot.get_guild(GUILD_ID)
-        after = guild.get_member(after.id)
-        await self.normalize_member(after)
+        for guild in self.bot.guilds:
+            after = guild.get_member(after.id)
```
Comment (pikaninja) [inline cogs/names.py]: again, what if None?

**Review by pikaninja — COMMENTED** [review]
Few issues and one random comment

File: cogs/moderation.py (hunk 2 above)
Comment (oliver-ni) [reply cogs/moderation.py]: s removed

File: cogs/moderation.py (hunk 3 above)
Comment (oliver-ni) [reply cogs/moderation.py:682]: its ok

**Pull request merged**